MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/MovieDetail/MovieDetailPresenter.swift

```diff
@@ -16,76 +15,64 @@ final class MovieDetailPresenter {
-            guard let self = self else { return }
-            
-            switch result {
-            case .success(let cast):
-                let mappedCast = cast.map { CastViewModel(fromModel: $0) }
-                self.delegate?.fillCastCV(with: mappedCast)
-            case .failure(let error):
-                print("Loading error: \(error.localizedDescription)")
-            }
-        }
+
+    private var mostPopularCast: AnyPublisher<[CastViewModel], Never> {
+        useCase
+            .getMostPopularCast(for: movieId)
+            .map { $0.map { CastViewModel(fromModel: $0) } }
+            .receiveOnMain()
     }
     
-    func getCrew() {
-        useCase.getCrew(for: movieId) { [weak self] result in
-            guard let self = self else { return }
-            
-            switch result {
-            case .success(let crew):
-                let mappedCrew = crew.map { CrewViewModel(fromModel: $0) }
-                self.delegate?.fillCrew(with: mappedCrew)
-            case .failure(let error):
-                print("Loading error: \(error.localizedDescription)")
-            }
-        }
+    private var crew: AnyPublisher<[CrewViewModel], Never> {
+        useCase
+            .getCrew(for: movieId)
+            .map { $0.map { CrewViewModel(fromModel: $0) } }
+            .receiveOnMain()
     }
     
-    func getRecommendations() {
-        useCase.getRecommendations(for: movieId) { [weak self] result in
-            guard let self = self else { return }
-            
-            switch result {
-            case .success(let movies):
-                let mappedMovies = movies.map { MovieViewModel(fromModel: $0) }
-                self.delegate?.fillRecommendationsCV(with: mappedMovies)
-            case .failure(let error):
-                print("Loading error: \(error.localizedDescription)")
-            }
-        }
+    private var recommendations: AnyPublisher<[MovieViewModel], Never> {
+        useCase
+            .getRecommendations(for: movieId)
+            .map { $0.map { MovieViewModel(fromModel: $0) } }
+            .receiveOnMain()
     }
     
-    func getReview() {
-        useCase.getReview(for: movieId) { [weak self] result in
-            guard let self = self else { return }
-            
-            switch result {
-            case .success(let review):
-                self.delegate?.fillReview(with: ReviewViewModel(fromModel: review))
-            case .failure(let error):
-                print("Loading error: \(error.localizedDescription). No review.")
-            }
-        }
+    private var review: AnyPublisher<ReviewViewModel?, Never> {
+        useCase
+            .fetchReviews(for: movieId)
+            .map { review in
+                guard let review = review else {
+                    return nil
+                }
+                
+                return ReviewViewModel(fromModel: review) }
+            .receiveOnMain()
```

Comment:
Generally, `receiveOnMain` is set on the variables that are used by the view controller, other `private` vars could be on background.
This is because you want to do as much work on the background thread and switch to main only when interacting with the views.

Reply:
Fixed [here](https://github.com/lihter/Movie-App/pull/48/commits/d4aa644216d84be7cc2bf830a9c58bff28410150#diff-d8a7df924093b6e29d479b7644fb3593ce7762a26b79b7f10f290852dfb0ce54R75).
Also fixed all of the other `private` vars.